includes/models/model.llms.user.certificate.php

```diff
@@ -752,6 +752,13 @@ private function set_property_defaults() {
 		$default_size                    = get_option( 'lifterlms_certificate_default_size', 'LETTER' );
 		$this->property_defaults['size'] = ! $default_size ? 'LETTER' : $default_size;
 
+		// Custom size.
+		if ( llms_parse_bool( get_option( 'lifterlms_certificate_default_custom_size', false ) ) ) {
+			$this->property_defaults['size']   = 'CUSTOM';
+			$this->property_defaults['width']  = get_option( 'lifterlms_certificate_default_custom_width', 8.5 );
+			$this->property_defaults['height'] = get_option( 'lifterlms_certificate_default_custom_height', 11 );
+			$this->property_defaults['unit']   = get_option( 'lifterlms_certificate_default_custom_unit', 'in' );
+		}
```

Comment:
My vision for this was that instead of replacing the `CUSTOM` definition with these default dimensions, it would register a new size definition using this filter: https://github.com/gocodebox/lifterlms/blob/8e9c2293977a14950275b3b9ca0d2284d051b804/includes/functions/llms.functions.certificate.php#L312-L364

The size would be `USER_DEFINED` which would mean the `CUSTOM` (that gets added outside this filter because it's not a defined / registered size) would still exist.

Your implementation does allow the custom size to be redefined per-certificate but in doing so it gets "detached" from the custom definition. If a user wishes to change their custom definition later, then they can't.

You've created a *default* value for the `CUSTOM` definition whereas I was planning for a UI that allows registration of a single custom size.

I also don't like the "leave checked to disable the default above" which is just kind of confusing to the user. I think if the `USER_DEFINED` is added (with pre-defined default values) and then users can use the settings below to modify the size settings it would be less confusing.

What do you think?

Reply:
Oh ok I get what you mean.
you're fucking smart!